File: problems/two_sum/TwoSum.java
package problems.two_sum;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    public static void main(String[] args) {
        int[] example = new int[] { 2, 1, 11, 7 };
        int[] expected = new int[] { 0, 3 };

        System.out.println("Expected output is : " + Arrays.toString(expected)
                + "output of solution is : " + Arrays.toString(twoSum(example, 9)));
    }

    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mapping = new HashMap<>();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            int current = nums[i];
            int complement = target - current;
            if (mapping.containsKey(complement)) {
                return new int[] { i, mapping.get(complement) };
            }
            mapping.put(current, i);
        }
        return new int[2];
    }
}

File: problems/two_sum/two_sum.py
def two_sum(nums: list[int], target: int):
    map = {}
    for index, value in enumerate(nums):
        current = value
        complement = target - current
        if complement in map:
            return [index, map[complement]]
        map[current] = index

    return []


print(two_sum([2, 7, 11, 15], 9))
